Sources/SwiftProtobuf/TextFormatScanner.swift

```diff
@@ -357,26 +272,153 @@ internal struct TextFormatScanner {
         return nil
     }
 
-    /// Assumes the leading quote has already been consumed
-    private mutating func parseQuotedString(terminator: UInt8) -> String? {
-        let start = p
-        while p != end {
-            let c = p[0]
-            if c == terminator {
-                let s = utf8ToString(bytes: start, count: p - start)
-                p += 1
-                skipWhitespace()
-                return s
+    /// Scan a string that encodes a byte field, return a count of
+    /// the number of bytes that should be decoded from it
+    private mutating func validateAndCountBytesFromString(terminator: UInt8) throws -> Int {
```

Comment:
So if you return a tuple (or overload and use a negative result), you could include a flag to say if there are any escape characters.  Then, if there are no escapes (likely the common case), parseBytesFromString can become a simple copy of that may bytes.  i.e. - since you pre scan, might as well collect a hint if you can optimize the later work.